src/tuner/algo/allreduce/ring.h

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_TUNER_ALLREDUCE_RING_H_
+#define NCCL_OFI_TUNER_ALLREDUCE_RING_H_
+
+#include "config.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "nccl-headers/nvidia/tuner.h"
+#include "./../../nccl_defaults.h"
```

Comment:
the top-level directory will have a -I in CPPFLAGS, so I'd have a slight preference for including this with a relative path from there (similar to other headers).

Reply:
moved it into `include/`.